artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPStandardMessage.java

```diff
@@ -162,14 +162,7 @@ public void reloadPersistence(ActiveMQBuffer record, CoreMessageObjectPools pool
       // Message state is now that the underlying buffer is loaded, but the contents not yet scanned
       resetMessageData();
       modified = false;
-      messageDataScanned = RELOAD_PERSISTENCE;
```

Comment:
really? that's the offending line? 👍

Reply:
from now on: there was already a method on AMQPMessage that will call reencode, if needed